### PR TITLE
Transition to pyproject.toml and setup.cfg

### LINKED ISSUE #1921: Transition Featuretools to use pyproject.toml and setup.cfg (moving away from setup.py)
> The future of Python packaging is pyproject.toml, and (for now) setup.cfg, based on [PEP 518](https://snarky.ca/clarifying-pep-518/) and (soon) [PEP 621](https://www.python.org/dev/peps/pep-0621/).

- http://ivory.idyll.org/blog/2021-transition-to-pyproject.toml-example.html
- Therefore we should migrate our Woodwork package to follow this new standard
- Woodwork steps (which should easily apply here): https://github.com/alteryx/woodwork/issues/1286

### PR DESCRIPTION
**Pull request by thehomebrewnerd**
### Transition to pyproject.toml and setup.cfg

Closes #1921

### CODE REVIEW

**Comment by gsheni** [comment]
Let's update our install test to run on on mac and ubuntu
```
name: Install Test
jobs:
  install_ft_complete:
    name: ${{ matrix.os }} - ${{ matrix.python_version }} install featuretools
    strategy:
      fail-fast: false
      matrix:
        python_version: ["3.8", "3.9"]
        os: [ubuntu-latest, macos-latest]
```

File: setup.cfg [hunk 1]
```diff
@@ -1,16 +1,139 @@
...
+* =
+    *.txt
+    LICENSE
+    README.md
+
+[options.exclude_package_data]
+* =
+    * __pycache__
+    *.py[co]
+
```
Comment (gsheni) [inline setup.cfg:119]:
```suggestion

[bdist_wheel]
universal = 1

```

**Comment by gsheni** [comment]
Also, we will need to update `install.md`:
- https://github.com/alteryx/featuretools/blob/main/docs/source/install.md#source

**Comment by thehomebrewnerd** [comment]
> Also, we will need to update `install.md`:
> 
> * https://github.com/alteryx/featuretools/blob/main/docs/source/install.md#source

@gsheni What needs to be updated in the install guide?

**Comment by gsheni** [comment]
One thing I forgot, the `minimum_dependency_checker.yml` file can be updated with v3.1
- https://github.com/alteryx/minimum-dependency-generator#setupcfg

**Review by gsheni — DISMISSED** [review]
lgtm

File: .github/workflows/minimum_dependency_checker.yml [hunk 2]
```diff
@@ -49,9 +50,11 @@ jobs:
           printf "${{ steps.min_dep_gen_core.outputs.min_reqs }}" > featuretools/tests/requirement_files/minimum_core_requirements.txt
       - name: Run min dep generator - koalas
         id: min_dep_gen_koalas
-        uses: alteryx/minimum-dependency-generator@v2
+        uses: alteryx/minimum-dependency-generator@v3.1
```
Comment (rwedge) [inline .github/workflows/minimum_dependency_checker.yml:53]: I think we'd want to use the "`v3` refers to the latest release in `v3.x`", similar to how other actions do tagging

We'd need to move the v3 tag to point at the 3.1 release and retag the current v3 as v3.0

Comment (gsheni) [reply .github/workflows/minimum_dependency_checker.yml:53]: No, it's fine. I control the release for that, and don't have any updates planned

Comment (rwedge) [reply .github/workflows/minimum_dependency_checker.yml:53]: It's fine as is but I think it'd be good practice to follow this convention

**Review by gsheni — APPROVED** [review]
Interesting. Installing in editable mode allowed you to get complete code coverage

**Comment by thehomebrewnerd** [comment]
> Interesting. Installing in editable mode allowed you to get complete code coverage

Yeah, don't fully understand the reason, but that was one difference I noticed as I reviewed the install commands.

**Pull request merged**